### PR TITLE
ex8_ch3.py

### PR DESCRIPTION
**Pull request by amymhaddad**
Instructions: Write a program to evenly divide pizzas. Prompt for the number of people, the number of pizzas, and the number of slices per pizza. Ensure that the number of pieces comes out even. Display the number of pieces of pizza each person should get. If there are leftovers, show the number of leftover pieces.

Example Output:
       How many people? 8
       How many pizzas do you have? 2
       8 people with 2 pizzas
       Each person gets 2 pieces of pizza.
       There are 0 leftover pieces.

### CODE REVIEW

File: ex8_ch3.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#step3: get input: how many slices are in a pizza?
+slices_per_pizza = int(input("How many slices are in a pizza? "))
+
+#step4: write calculation to see how many total slices there are
+total_pizza_slices = total_pizzas * slices_per_pizza
+
+#step5: write calculation to divide the pizza slices evenly among the total people
+slices_per_person = total_pizza_slices / pizza_eaters
+
+print(f"Each person can have {int(slices_per_person)} pieces of pizza.")
```
Comment (amymhaddad) [inline ex8_ch3.py]: Question: I had to add "int" here. If I don't, then I get a float. Why is that? Based on all of the variables above, this line of code should also be an int -- correct?

Comment (paulghaddad) [reply ex8_ch3.py]: It appears that in Python 3, the `/` operator results in true division, which means a `float` is always returned. There is a `int` division, the truncates the decimal: `//`.

I'm thinking the `//` would work here and should give you the same answer.

I actually had to look this up myself!

Here's a good article: http://www.informit.com/articles/article.aspx?p=1439189

**Comment by amymhaddad** [comment]
@paulghaddad I updated this exercise and used floor division based on your recommendation. However, is floor division the most accurate for this exercise? Floor division rounds up, so would it be better to use true division and then solve for the remainder?

File: ex8_ch3.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+@author: amyhaddad
+"""
+
+#step1: get input
+number_people = int(input("Enter the number of people: "))
+total_pizzas = int(input("Enter the total number of pizzas: "))
+slices_per_pizza = int(input("Enter the slices per pizza: "))
+
+#step2: calculate the total slices of pizza
+total_slices = total_pizzas * slices_per_pizza
```
Comment (paulghaddad) [inline ex8_ch3.py]: Nice intermediate step!

File: ex8_ch3.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+#step3: calucate the slices per person
+slices_per_person = total_slices // number_people
```
Comment (paulghaddad) [inline ex8_ch3.py]: I think you want floor division. This divides the pieces into whole slices. It doesn't look like the problem is concerned with the it being fractional, like 2.5.

Comment (paulghaddad) [reply ex8_ch3.py]: However, it looks like the problem wants you to print out the remaining pieces as well as _whole_ pieces_ per person. So in this case you can use the modulo operator to find that.

**Comment by amymhaddad** [comment]
@paulghaddad I updated this exercise

**Pull request closed without merging**